docs/api.yaml

```diff
@@ -0,0 +1,1061 @@
+
+    Student:
+      type: object
+      properties:
+        id:
+          type: string
+        first_name:
+          type: string
+        last_name:
+          type: string
+        sex:
+          type: string
+          enum:
+            - M
+            - F
+        birth_date:
+          type: "string"
+          format: "date"
+        address:
+          type : string
+        phone_number:
+          type: string
+        email:
+          type: string
+        entrance_date:
+          type: "string"
+          format: "date"
+        status:
+          type: string
+          description: "Defaults to ENABLED"
+          enum:
+            - ENABLED
+            - DISABLED
+        group:
+          $ref: '#/components/schemas/Group'
+        enrolled_courses:
+          type: array
+          items:
+            $ref: '#/components/schemas/EnrolledCourse'
+        grades:
+          type: array
+          items:
+            $ref: '#/components/schemas/Grade'
+
+    CreateStudent:
+      type: object
+      properties:
+        first_name:
+          type: string
+        last_name:
+          type: string
+        sex:
+          type: string
+          enum:
+            - M
+            - F
+        birth_date:
+          type: "string"
+          format: "date"
+        address:
+          type : string
+        phone_number:
+          type: string
+        email:
+          type: string
+        entrance_date:
+          type: "string"
+          format: "date"
+        status:
+          type: string
+          enum:
+            - ACTIVATED
+            - DEACTIVATED
+        group:
+          $ref: '#/components/schemas/Group'
+        enrolled_courses:
+          type: array
+          items:
+            $ref: '#/components/schemas/EnrolledCourse'
+        grades:
+          type: array
+          items:
+            $ref: '#/components/schemas/Grade'
+
+    ######################################################
+    ####################### TEACHER ######################
+    ######################################################
+
+    CreateTeacher:
+      type: object
+      properties:
+        first_name:
+          type: string
+        last_name:
+          type: string
+        sex:
+          type: string
+          enum:
+            - M
+            - F
+        address:
+          type : string
+        phone_number:
+          type: integer
+        email:
+          type: string
+        mariage_status:
+          type: string
+        status:
+          type: string
+          description: "Defaults to ENABLED"
+          enum:
+            - ENABLED
+            - DISABLED
+        birth_date:
+          type: "string"
+          format: "date"
+        hire_date:
+          type: "string"
+          format: "date"
+        taught_courses:
+          type: array
+          items:
+            $ref: "#/components/schemas/TaughtCourse"
+        interventions:
```

Comment:
What's the use of this? What's the difference with TaughtCourse.timetable?
Again, student.timetable is the class hours student must attend to. teacher.timetable is about the class hours the teacher must deliver. course.timetable is the union of all teacher.timetable

Again, if you're not comfortable with using the same name but having different meaning depending on the context, prefix the name with some context

Reply:
The attribute required_teaching_hours is the difference with the TaughtCourse and the TaughtCourse.timetable (TaughtPeriod).
As you said, a course has a main teacher, but other teachers can give the same course so when you added this attribute I think it's to know how many hours is required by each teachers.

I do agree with you with the course.timetable, but we didn't add this attribute _(yet)_ because this feature has not _(yet)_ been defined in the administration profile.